flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointTestBase.java

```diff
@@ -221,15 +228,17 @@ public InputStatus pollNext(ReaderOutput<Long> output) throws InterruptedExcepti
                 output.collect(withHeader(split.nextNumber), split.nextNumber);
                 split.nextNumber += split.increment;
 
-                if (throttle) {
-                    // throttle source as long as sink is not backpressuring (which it does only
-                    // after full recovery)
-                    Thread.sleep(1);
+                switch (pollingState) {
+                    case FINISHING:
+                        return InputStatus.END_OF_INPUT;
+                    case THROTTLING:
+                        // throttle source as long as sink is not backpressuring (which it does only
+                        // after full recovery)
+                        Thread.sleep(1);
+                        return InputStatus.MORE_AVAILABLE;
+                    default:
+                        return InputStatus.MORE_AVAILABLE;
```

Comment:
nitty nit: add `case PUMPING` and make `default: throw new UnssuportedOperationException(pollingState);`?